### DISCUSSION TITLE
A research for generating PR checklists in Pull Request Template

### DISCUSSION THREAD

**Discussion by baishuotong** [PyGithub/PyGithub#2849] [Show and tell]
Dear maintainers and communities interested in this project:

Our team is currently developing a tool for automatically generating PR checklists in Pull Request Template. The tool generates a set of PR rules based on the contribution guidelines in your project, which external contributors may be expected to adhere to. Below are the results of our generation:

**Test**: 
1. Automated test through pytest. 
2. Run pytest k Issue139 test Completion record, pytest Issue139 test Completion record auth_with_token, pytest Issue139.testCompletion record auth_with_jwt. 

**Documentation**: 
1. Build documentation locally.
2. The problem to solve and how the author went about solving the problem

**Style**: 
1. Adopts the black coding style, tox -e lint.
2. let [pre-commit](https://pre-commit.com/#installation) fix, when forgetting coding style

Also, we suggest you add some checklist notes about **Branch**, **Title**, **commit message**, **Description**, **contributing**, and **CLA** which are also important for submitting PRs according to our analysis.
We hope you and your team can provide feedback on our generated output.

Best regards.

**Comment by BerylYanjie** [comment]
I can accept notes reflected in the contributing.md, but I won't consider adding items that haven't appeared before. I feel it's unnecessary
